File: CzhERP.Application/Entity/StoInventory.cs
namespace CzhERP.Application.Entity;

[Tenant("1300000000001")]
[SugarTable("Sto_Inventory", "库存余额")]
public partial class StoInventory : EntityBase
{
    [Required]
    [SugarColumn(ColumnName = "WarehouseId", ColumnDescription = "仓库ID")]
    public virtual long WarehouseId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "WarehouseCode", ColumnDescription = "仓库编码", Length = 50)]
    public virtual string WarehouseCode { get; set; }

    [SugarColumn(ColumnName = "WarehouseName", ColumnDescription = "仓库名称", Length = 100)]
    public virtual string? WarehouseName { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialId", ColumnDescription = "物料ID")]
    public virtual long MaterialId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialCode", ColumnDescription = "物料编码", Length = 50)]
    public virtual string MaterialCode { get; set; }

    [SugarColumn(ColumnName = "MaterialName", ColumnDescription = "物料名称", Length = 100)]
    public virtual string? MaterialName { get; set; }

    [SugarColumn(ColumnName = "Spec", ColumnDescription = "规格型号", Length = 100)]
    public virtual string? Spec { get; set; }

    [Required]
    [SugarColumn(ColumnName = "Unit", ColumnDescription = "单位", Length = 20, DefaultValue = "个")]
    public virtual string Unit { get; set; } = "个";

    [SugarColumn(ColumnName = "StockQuantity", ColumnDescription = "库存数量", DefaultValue = "0")]
    public virtual decimal StockQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "FrozenQuantity", ColumnDescription = "冻结数量", DefaultValue = "0")]
    public virtual decimal FrozenQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "AvailableQuantity", ColumnDescription = "可用数量", DefaultValue = "0")]
    public virtual decimal AvailableQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "CostPrice", ColumnDescription = "成本单价", DefaultValue = "0")]
    public virtual decimal CostPrice { get; set; } = 0;

    [SugarColumn(ColumnName = "TotalCost", ColumnDescription = "库存总成本", DefaultValue = "0")]
    public virtual decimal TotalCost { get; set; } = 0;

    [SugarColumn(ColumnName = "MinStock", ColumnDescription = "最低库存", DefaultValue = "0")]
    public virtual decimal MinStock { get; set; } = 0;

    [SugarColumn(ColumnName = "MaxStock", ColumnDescription = "最高库存", DefaultValue = "0")]
    public virtual decimal MaxStock { get; set; } = 0;

    [SugarColumn(ColumnName = "UpdateTime", ColumnDescription = "更新时间")]
    public virtual DateTime? UpdateTime { get; set; }
}

File: Web/src/api/stock/stoInventory.ts
import {useBaseApi} from '/@/api/base';

// 库存余额接口服务
export const useStoInventoryApi = () => {
	const baseApi = useBaseApi("stoInventory");
	return {
		// 分页查询库存余额
		page: baseApi.page,
		// 查看库存余额详细
		detail: baseApi.detail,
		// 新增库存余额
		add: baseApi.add,
		// 更新库存余额
		update: baseApi.update,
		// 删除库存余额
		delete: baseApi.delete,
		// 批量删除库存余额
		batchDelete: baseApi.batchDelete,
		// 导出库存余额数据
		exportData: baseApi.exportData,
		// 导入库存余额数据
		importData: baseApi.importData,
		// 下载库存余额数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 库存余额实体
export interface StoInventory {
	// 主键Id
	id: number;
	// 仓库ID
	warehouseId?: number;
	// 仓库编码
	warehouseCode?: string;
	// 仓库名称
	warehouseName: string;
	// 物料ID
	materialId?: number;
	// 物料编码
	materialCode?: string;
	// 物料名称
	materialName: string;
	// 规格型号
	spec: string;
	// 单位
	unit?: string;
	// 库存数量
	stockQuantity?: number;
	// 冻结数量
	frozenQuantity?: number;
	// 可用数量
	availableQuantity?: number;
	// 成本单价
	costPrice?: number;
	// 库存总成本
	totalCost?: number;
	// 最低库存
	minStock?: number;
	// 最高库存
	maxStock?: number;
	// 更新时间
	updateTime: string;
	// 创建时间
	createTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}